### PR TITLE
REGR: exceptions not caught in _call_map_locations

### PR DESCRIPTION
**Pull request by simonjayhawkins**
- [ ] xref #32409
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

# NOTE: issue 'fixed' on master. This PR against 1.0.x branch

cc @jbrockmendel Could there still be issue on master with the exceptions not being caught?

### CODE REVIEW

File: pandas/_libs/index.pyx [hunk 1]
```diff
@@ -509,7 +509,7 @@ cdef class PeriodEngine(Int64Engine):
     cdef _get_index_values(self):
         return super(PeriodEngine, self).vgetter()
 
-    cdef void _call_map_locations(self, values):
+    cpdef _call_map_locations(self, values):
```
Comment (jbrockmendel) [inline pandas/_libs/index.pyx:512]: these are still cdef on master, why is this the appropriate fix on 1.0.4?

Comment (simonjayhawkins) [reply pandas/_libs/index.pyx:512]: > why is this the appropriate fix on 1.0.4?

#29232 was a clean-up PR which caused a regression. this is reverting those changes. (applied directly to 1.0.x branch and not touching master, see https://github.com/pandas-dev/pandas/pull/34048#issuecomment-625347097)

> these are still cdef on master

that's my question to you in the OP.

> Could there still be issue on master with the exceptions not being caught?

Comment (jbrockmendel) [reply pandas/_libs/index.pyx:512]: > Could there still be issue on master with the exceptions not being caught?

I can't reproduce the bugs from #32409.  I'm guessing you bisected to find the commit that introduced (reintroduced?) the bug, did you find what commit fixed it?

Comment (simonjayhawkins) [reply pandas/_libs/index.pyx:512]: yep fixed on master, see https://github.com/pandas-dev/pandas/issues/32409#issuecomment-606627286

but not sure whether the PR that fixed it #31294 should be backported. I can't immediately see the relation. 

I see this PR as the safer option as it's a simple revert.

Comment (jbrockmendel) [reply pandas/_libs/index.pyx:512]: > I see this PR as the safer option as it's a simple revert.

Works for me

**Pull request merged**